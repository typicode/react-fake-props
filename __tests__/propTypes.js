import path from 'path'
import fakeProps from '../src'

const ComponentFile = path.resolve(
  './fixtures/propTypes/Component.jsx'
)
const SimpleFile = path.resolve('./fixtures/propTypes/Simple.jsx')
const NoProps = path.resolve('./fixtures/propTypes/NoProps.jsx')
const Multiple = path.resolve('./fixtures/propTypes/Multiple.jsx')
const NestedShape = path.resolve('./fixtures/propTypes/NestedShape.jsx')

describe('fakeProps', () => {
  it('should return an object with no props (snapshot)', () => {
    expect(fakeProps(NoProps)).toMatchSnapshot()
  })

  it('should return an object with required props faked (snapshot)', () => {
    expect(fakeProps(ComponentFile)).toMatchSnapshot()
  })

  it('should return an object with all props faked (snapshot)', () => {
    expect(fakeProps(ComponentFile, { optional: true })).toMatchSnapshot()
  })

  it('should work with nested shape objects', () => {
    expect(fakeProps(NestedShape, { optional: true })).toMatchSnapshot()
  })

  it('should return more props with optional', () => {
    const allProps = fakeProps(ComponentFile, { optional: true })
    const requiredProps = fakeProps(ComponentFile)

    const allPropsLength = Object.keys(allProps).length
    const requiredPropsLength = Object.keys(requiredProps).length

    expect(allPropsLength).toBeGreaterThan(requiredPropsLength)
  })

  it('should return an object with shape props faked', () => {
    const props = fakeProps(SimpleFile)
    const { requiredString, requiredShape } = props

    expect(props).toEqual({
      requiredString,
      requiredShape: {
        requiredString: requiredShape.requiredString
      }
    })
  })

  it('should prefix shape props', () => {
    const props = fakeProps(SimpleFile)
    expect(props.requiredShape.requiredString).toBe(
      'requiredShape.requiredString'
    )
  })

  it('should work with multiple components', () => {
    const component = fakeProps(Multiple, { all: true }).find(
      ({ displayName }) => displayName === 'First'
    )
    expect(component.props.requiredString).toBe('requiredString')
  })
})
