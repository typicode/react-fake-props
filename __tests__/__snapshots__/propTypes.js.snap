// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fakeProps should return an object with all props faked (snapshot) 1`] = `
Object {
  "arrayOfObjects": Array [
    Object {
      "age": 1,
      "isGood": true,
      "name": "arrayOfObjects.name",
    },
  ],
  "customArrayProp": Array [
    "custom type not supported, please set the correct value for customArrayProp prop",
  ],
  "customProp": "custom type not supported, please set the correct value for customProp prop",
  "optionalArray": Array [],
  "optionalArrayOf": Array [
    1,
  ],
  "optionalBool": true,
  "optionalElement": <div>
    fake optionalElement element
</div>,
  "optionalEnum": "'News'",
  "optionalFunc": [Function],
  "optionalMessage": "instanceOf type not supported, please set the correct value for optionalMessage prop",
  "optionalNode": "optionalNode",
  "optionalNumber": 1,
  "optionalObject": Object {},
  "optionalObjectOf": Object {
    "prop": 1,
  },
  "optionalObjectWithShape": Object {
    "color": "optionalObjectWithShape.color",
    "fontSize": 1,
  },
  "optionalString": "optionalString",
  "optionalSymbol": Symbol(),
  "optionalUnion": "optionalUnion",
  "requiredAny": "any",
  "requiredFunc": [Function],
}
`;

exports[`fakeProps should return an object with required props faked (snapshot) 1`] = `
Object {
  "requiredAny": "any",
  "requiredFunc": [Function],
}
`;
